plugins/omdtls/omdtls.c

```diff
@@ -0,0 +1,885 @@
+			  "param '%s' in beginCnfLoad\n", modpblk.descr[i].name);
+		}
+	}
+finalize_it:
+	if(pvals != NULL)
+		cnfparamvalsDestruct(pvals, &modpblk);
+ENDsetModCnf
+
+BEGINendCnfLoad
+CODESTARTendCnfLoad
+	loadModConf = NULL; /* done loading */
+ENDendCnfLoad
+
+BEGINcheckCnf
+CODESTARTcheckCnf
+ENDcheckCnf
+
+BEGINactivateCnfPrePrivDrop
+	instanceData *inst;
+CODESTARTactivateCnfPrePrivDrop
+	runModConf = pModConf;
+	DBGPRINTF("omdtls: activateCnfPrePrivDrop\n");
+
+	for(inst = runModConf->root ; inst != NULL ; inst = inst->next) {
+		// Init OpenSSL Context with DTLS_method
+		CHKiRet(net_ossl.osslCtxInit(inst->pNetOssl, DTLS_method()));
+	}
+finalize_it:
+ENDactivateCnfPrePrivDrop
+
+BEGINactivateCnf
+CODESTARTactivateCnf
+ENDactivateCnf
+
+BEGINfreeCnf
+CODESTARTfreeCnf
+	free(pModConf->tplName);
+ENDfreeCnf
+
+BEGINcreateInstance
+CODESTARTcreateInstance
+	DBGPRINTF("createInstance[%p]: ENTER\n", pData);
+	pData->tplName = NULL;
+	pData->target = NULL;
+	pData->port = NULL;
+	pData->authMode = OSSL_AUTH_CERTANON;
+	pData->tlscfgcmd = NULL;
+	pData->statsName = NULL;
+
+	/* node created, let's add to config */
+	if(loadModConf->tail == NULL) {
+		loadModConf->tail = loadModConf->root = pData;
+	} else {
+		loadModConf->tail->next = pData;
+		loadModConf->tail = pData;
+	}
+
+	// Construct pNetOssl helper
+	CHKiRet(net_ossl.Construct(&pData->pNetOssl));
+finalize_it:
+ENDcreateInstance
+
+
+BEGINcreateWrkrInstance
+CODESTARTcreateWrkrInstance
+	DBGPRINTF("createWrkrInstance[%p]: ENTER\n", pWrkrData);
+	pWrkrData->bIsConnecting = 0;
+	pWrkrData->bIsConnected = 0;
+	pWrkrData->bIsSuspended = 0;
+
+	CHKiRet(pthread_rwlock_init(&pWrkrData->pnLock, NULL));
+
+	pWrkrData->bThreadRunning = 0;
+	pWrkrData->tid = 0;
+
+	// Run Proton Worker Thread now
+//	dtls_run_thread(pWrkrData);
+finalize_it:
+	DBGPRINTF("createWrkrInstance[%p] returned %d\n", pWrkrData, iRet);
+ENDcreateWrkrInstance
+
+
+BEGINisCompatibleWithFeature
+CODESTARTisCompatibleWithFeature
+ENDisCompatibleWithFeature
+
+
+BEGINfreeInstance
+CODESTARTfreeInstance
+	DBGPRINTF("freeInstance[%p]: ENTER\n", pData);
+
+	if (pData->stats) {
+		statsobj.Destruct(&pData->stats);
+	}
+
+	// DeConstruct pNetOssl helper
+	net_ossl.Destruct(&pData->pNetOssl);
+
+	/* Free other mem */
+	free(pData->target);
+	free(pData->port);
+	free(pData->tlscfgcmd);
+
+	free(pData->tplName);
+	free(pData->statsName);
+ENDfreeInstance
+
+BEGINfreeWrkrInstance
+CODESTARTfreeWrkrInstance
+	DBGPRINTF("freeWrkrInstance[%p]: ENTER\n", pWrkrData);
+
+	/* Closing azure first! */
+	pthread_rwlock_wrlock(&pWrkrData->pnLock);
+
+	// Close Proton Connection
+	dtls_close(pWrkrData);
+
+	// Stop Proton Handle Thread
+//	dtls_shutdown_thread(pWrkrData);
+
+	pthread_rwlock_unlock(&pWrkrData->pnLock);
+
+	pthread_rwlock_destroy(&pWrkrData->pnLock);
+ENDfreeWrkrInstance
+
+
+BEGINdbgPrintInstInfo
+CODESTARTdbgPrintInstInfo
+ENDdbgPrintInstInfo
+
+BEGINtryResume
+CODESTARTtryResume
+#ifndef NDEBUG
```

Comment:
I would tend to do the dbgprintf also in production builds. Makes troubleshooting easier

Reply:
what's special in this case?